isn_tractor/ibisn.py

```diff
@@ -284,7 +287,12 @@ def __make_array(*xs):
     return np.array(xs, dtype=object)
 
 
-def __make_edge_fn(data, metric_fn: MetricFn, pool_fn: PoolingFn):
+def __make_edge_fn(
+    data,
+    metric_fn: MetricFn,
+    pool_fn: PoolingFn,
```

Comment:
I have the feeling that the __pooling function - which I can't find anymore but probably it's just me - is what is making the tests fail, but may be just a stupid hypothesis.

Reply:
:) it's right there where you've highlighted: `pool_fn`.

Reply:
I didn't express myself correctly; what I meant is that it's not defined before; it's like we are calling the pooling function when there's no pooling function defined. Dunno if I was clearer now - and dunno if that's even the actual problem tho.

Reply:
Ah I see. https://github.com/GiadaLalli/ISN-tractor/blob/cuda_impl/isn_tractor/ibisn.py#L358